build_tools/ci/run_matmul_test.sh

```diff
@@ -770,6 +770,14 @@ run_matmul_test \
     --acc_type "i32" \
     --m "1536" --k "2048" --n "1536"
 
+run_matmul_test \
+    --name_prefix "small" \
+    --lower_to_aie_pipeline "objectFifo" \
+    --tile_pipeline "pack-peel" \
+    --lhs_rhs_type "bf16" \
+    --acc_type "f32" \
+    --m "64" --k "64" --n "64"
```

Comment:
Hi @yzhang93,

Yes, you're indeed correct.

I've logged [128x128x256 findings here](https://github.com/nod-ai/iree-amd-aie/issues/431#issuecomment-2197181642).

The intention was then to first add support in order to make basic/small shape bf16 Matmul work on objectfifo.

With @jtuyls' support for varied shapes on i32 and after this PR lands, we can take a look at supporting other shapes for bf16 incrementally.

Reply:
The log is quite old. Does it still have the same error? I'm asking because I'm not sure if the casting you implemented in this PR work in general or just work because the starting offsets are all 0.

Reply:
Oh yes yes - it's still the same in `AIEAssignBufferAddresses` pass.